pkg/ovs/ovsctl/appctl.go

```diff
@@ -75,18 +76,30 @@ func (c *ovsCtlClient) Trace(req *TracingRequest) (string, error) {
 	}
 
 	if req.SrcIP != nil {
-		if strings.Contains(req.Flow, "nw_src=") {
-			return "", newBadRequestError("duplicated 'nw_src' in flow")
+		nwSrcKey := getNwSrcKey(req.SrcIP)
+		if strings.Contains(req.Flow, fmt.Sprintf("%s=", nwSrcKey)) {
+			return "", newBadRequestError(fmt.Sprintf("duplicated '%s' in flow", nwSrcKey))
 		} else {
-			nwSrc = fmt.Sprintf("nw_src=%s,", req.SrcIP.String())
+			nwSrc = fmt.Sprintf("%s=%s,", nwSrcKey, req.SrcIP.String())
+		}
+		if strings.Contains(nwSrcKey, "ipv6") {
+			ip = "ipv6"
+		} else {
+			ip = "ip"
 		}
 	}
 	if req.DstIP != nil {
```

Comment:
Do we check source and dest are the same family somewhere?

Reply:
If the src/dst IP is input in the command directly, we have a validation. If user only inputs the Pod name/namespace in the command, we are using the same address family to retrieve the IP addresses, and if we can't read the IP with the given address family, an error would return.